components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/dao/ApiMgtDAO.java

```diff
@@ -14723,4 +14739,246 @@ public String getUserID(String userName) throws APIManagementException {
         }
         return list;
     }
+
+    /**
+     * Remove the Pending workflow Requests using ExternalWorkflowReference
+     *
+     * @param workflowExtRef External Workflow Reference of Workflow Pending Request
+     * @throws APIManagementException
+     */
+    public void deleteWorkflowRequest(String workflowExtRef) throws APIManagementException {
+        Connection connection = null;
+        PreparedStatement prepStmt = null;
+        String query = SQLConstants.DELETE_WORKFLOW_REQUEST_SQL;
+        try {
+            connection = APIMgtDBUtil.getConnection();
+            connection.setAutoCommit(false);
+            prepStmt = connection.prepareStatement(query);
+            prepStmt.setString(1, workflowExtRef);
+            prepStmt.executeUpdate();
+            connection.commit();
+        } catch (SQLException e) {
+            handleException("Failed to delete the workflow request. ", e);
+        } finally {
+            APIMgtDBUtil.closeAllConnections(prepStmt, connection, null);
+        }
```

Comment:
Use "try with resources" whenever possible. We can do the same functionality in less amount of code.